### PR TITLE
FATES api 8, nutrients

### PR DESCRIPTION
**Pull request by rgknox**
This changeset updates the interface with FATES, which accommodates nutrient fluxes through FATES and into the ELM land model.  Changes are backwards incompatible, FATES had a major api update.   This included changes to calls in the clmfates_interface, as well as code handling history writing. Notably, a change was added that allows history variables to exist, with a second dimension of size 1. Previously, the code would not allow this condition.

[non-BFB] for cases with FATES active
[BFB] for all other cases

### CODE REVIEW

**Comment by rgknox** [comment]
rebased changes off master

**Comment by dqwu** [comment]
This PR (merged to next branch) has caused the following e3sm_developer test to fail on anlworkstation:
ERS_Ld3.f45_f45.ICLM45ED.anlworkstation_gnu.clm-fates

Error messages in e3sm.log:
```
...
clmfates_interfaceMod.F90:: reading fates_seed_suppl
ENDRUN:clmfates_interface::ReadParametereadNcdio:: - Error reading.  Var:fates_seed_suppl
ERROR: Unknown error submitted to shr_abort_abort.
```

**Comment by rgknox** [comment]
@dqwu , thanks for the feedback.  I'm now realizing that with this update I will have to update the default FATES parameter file.

**Comment by jqyin** [comment]
re-merged to next to fix fates test

**Comment by wlin7** [comment]
@jqyin , do you think it ready to re-merge the PR?

**Comment by rgknox** [comment]
@wlin7 , I will let @jqyin respond as well.  But I forgot to mention that the land_developer test suite does pass on cori now.

File: components/clm/src/main/histFileMod.F90 [hunk 1]
```diff
@@ -983,17 +989,19 @@ subroutine hist_update_hbuf(bounds)
...
-          num2d = tape(t)%hlist(f)%field%num2d
-          if ( num2d == 1) then
+          numdims = tape(t)%hlist(f)%field%numdims
+          if ( numdims == 1) then   
              call hist_update_hbuf_field_1d (t, f, bounds)
           else
+             num2d = tape(t)%hlist(f)%field%num2d
              call hist_update_hbuf_field_2d (t, f, bounds, num2d)
           end if
        end do
```
Comment (amametjanov) [inline components/clm/src/main/histFileMod.F90]: `numdims` is a local variable that can be overwritten by multiple threads (race hazard). I'll push an update:
```
-!$OMP PARALLEL DO PRIVATE (f, num2d)
+!$OMP PARALLEL DO PRIVATE (f, numdims, num2d)
        do f = 1,tape(t)%nflds
           numdims = tape(t)%hlist(f)%field%numdims
           if ( numdims == 1) then
```

**Pull request merged**